plugins/dynamic-few-shot-lambda/template.yml

```diff
@@ -0,0 +1,216 @@
+              Resource:
+                - !Ref DynamicFewShotVectorIndex
+            - Effect: Allow
+              Action:
+                - kms:Encrypt
+                - kms:Decrypt
+                - kms:ReEncrypt*
+                - kms:GenerateDataKey*
+                - kms:DescribeKey
+              Resource:
+                - !Ref GenAIIDPCustomerManagedEncryptionKeyArn
+
+  DynamicFewShotLogGroup:
+    Type: AWS::Logs::LogGroup
+    Metadata:
+      cfn_nag:
+        rules_to_suppress:
+          - id: W84
+            reason: "Demo function - KMS CMK not required, but can be added by customer for production use cases"
+    # checkov:skip=CKV_AWS_158: "Demo function - KMS CMK not required, but can be added by customer for production use cases"
+    Properties:
+      LogGroupName: !Sub "/aws/lambda/${LambdaFunctionName}"
+      RetentionInDays: 7  # Short retention for demo purposes
+
+  DynamicFewShotVectorBucket:
+    Type: AWS::S3Vectors::VectorBucket
+    Metadata:
+      cfn_nag:
+        rules_to_suppress:
+          - id: W84
+            reason: "Demo function - KMS CMK not required, but can be added by customer for production use cases"
+    # checkov:skip=CKV_AWS_158: "Demo function - KMS CMK not required, but can be added by customer for production use cases"
+    Properties:
+      VectorBucketName: !Ref VectorBucketName
+      EncryptionConfiguration:
+        SseType: "AES256"
+
+  DynamicFewShotVectorIndex:
+    Type: AWS::S3Vectors::Index
+    Properties:
+      IndexName: !Ref VectorIndexName
+      DataType: "float32"
+      Dimension: !Ref VectorDimensions
+      DistanceMetric: "cosine"
+      MetadataConfiguration:
+        NonFilterableMetadataKeys:
+          - "classPrompt"
+          - "attributesPrompt"
+          - "imagePath"
+      VectorBucketArn: !Ref DynamicFewShotVectorBucket
+
+Outputs:
+
+  DynamicFewShotFunctionName:
+    Description: Name of the demo Lambda function
+    Value: !Ref DynamicFewShotFunction
+
+  DynamicFewShotFunctionArn:
+    Description: ARN of the demo Lambda function (use this in your GenAIIDP configuration)
+    Value: !GetAtt DynamicFewShotFunction.Arn
+
+  DynamicFewShotLogGroup:
+    Description: CloudWatch Log Group for monitoring demo Lambda execution
+    Value: !Ref DynamicFewShotLogGroup
+
+  DynamicFewShotVectorBucketArn:
+    Description: S3 Vectors bucket for dynamic few-shot examples
+    Value: !Ref DynamicFewShotVectorBucket
+
+  DynamicFewShotVectorIndexArn:
+    Description: S3 Vectors index for dynamic few-shot examples
+    Value: !Ref DynamicFewShotVectorIndex
+
+  DynamicFewShotDatasetBucket:
+    Description: S3 Bucket for example data sets
+    Value: !Ref DynamicFewShotDatasetBucket
+
+  UsageInstructions:
+    Description: How to use this Lambda in your IDP configuration
+    Value: !Sub |
+      Add this ARN to your extraction config:
+      extraction:
+        dynamic_few_shot_lambda_arn: "${DynamicFewShotFunction.Arn}"
+        
+  MonitoringLink:
```

Comment:
Nice touch..  A CW Dashboard could be even better, to track number of inferences, successful matches, throttles/retries, etc.